app/src/main/java/com/ncnf/database/builder/DatabaseObjectBuilder.java

```diff
@@ -0,0 +1,10 @@
+package com.ncnf.database.builder;
+
+import java.util.Map;
+
+public abstract class DatabaseObjectBuilder<T> {
```

Comment:
Could it be an Interface ? Because it does not provide any implementation, only signatures.

Reply:
+1

Reply:
I think it was a problem of genericity which lead to this implementation.